sorts/Linear_Time_Sort.py

```diff
@@ -0,0 +1,63 @@
+# Python3 the implementation to sort an  
+# array of size n 
+  
+# A function to do counting sort of arr[]  
+# according to the digit represented by exp.  
+def countSort(arr, n, exp):  
+    output = [0] * n # output array  
+    count = [0] * n 
+    for i in range(n): 
+        count[i] = 0
+  
+    # Store count of occurrences in count[]  
+    for i in range(n): 
+        count[ (arr[i] // exp) % n ] += 1
+  
+    # Change count[i] so that count[i] now contains  
+    # actual position of this digit in output[]  
+    for i in range(1, n):  
+        count[i] += count[i - 1]  
+  
+    # Build the output array  
+    for i in range(n - 1, -1, -1):  
+  
+        output[count[ (arr[i] // exp) % n] - 1] = arr[i]  
+        count[(arr[i] // exp) % n] -= 1
+  
+    # Copy the output array to arr[], so that  
+    # arr[] now contains sorted numbers according  
+    # to current digit  
+    for i in range(n):  
+        arr[i] = output[i]  
+  
+# The main function to that sorts arr[] of  
+# size n using Radix Sort  
+def sort(arr, n) : 
+      
```

Comment:
Let’s move line 53 (n = len(arr)) to here.  The caller only needs to give us an array and we can get its length ourselves.

Please add doctests to both functions.